### PR TITLE
Convert Prometheus Exporter to Pull MetricReader

### LINKED ISSUE #1941: prometheus duplicate metrics
prometheus metrics endpoint return duplicate metrics and not updating them properly on a [SimpleWebServer](https://gitlab.com/eidheim/Simple-Web-Server) api endpoints.

**Steps to reproduce**
Created a sample repo which reproduces this issue. https://github.com/jayasai470/opentelemetry-cpp-sample

**What is the expected behavior?**
Get metrics without duplicates

**What is the actual behavior?**
```
# HELP http_server_duration_count 1.8.1
# TYPE http_server_duration_count counter
http_server_duration_count{method="GET",status="200",path="/api/health"} 3 1674066565733
http_server_duration_count{method="GET",status="200",path="/api/test"} 2 1674066565733
# HELP http_server_duration_count 1.8.1
# TYPE http_server_duration_count counter
http_server_duration_count{method="GET",status="200",path="/api/test"} 2 1674066566733
http_server_duration_count{method="GET",status="200",path="/api/health"} 3 1674066566733
```

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #1941

## Changes
Changes are as explained here - https://github.com/open-telemetry/opentelemetry-cpp/issues/1941#issuecomment-1413236782

Prometheus Exporter now implements `MetricReader` class, instead of `PushMetricExporter`. And the metrics is collected only when the actual http pull request is received from Prometheus  agent to fetch the metrics. Earlier metrics were fetched periodically by PeriodicMetricReader and stored into intermediate storage, from where the Prometheus agent used to pull the data. This resulted in duplicate entries of data in the intermediate storage.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
This is ready for review now. @esigo - you know this code best, assigning to you :)

File: exporters/prometheus/test/exporter_utils_test.cc [hunk 1]
```diff
@@ -81,18 +81,21 @@ void assert_histogram(prometheus_client::MetricFamily &metric,
...
 TEST(PrometheusExporterUtils, TranslateToPrometheusEmptyInputReturnsEmptyCollection)
 {
-  auto translated = PrometheusExporterUtils::TranslateToPrometheus(
-      std::vector<std::unique_ptr<metric_sdk::ResourceMetrics>>{});
+  metric_sdk::ResourceMetrics metrics_data = {};
+  auto translated = PrometheusExporterUtils::TranslateToPrometheus(metrics_data);
   ASSERT_EQ(translated.size(), 0);
 }
 
+#if 0
```
Comment (esigo) [inline exporters/prometheus/test/exporter_utils_test.cc:89]: Would it possible to change the unit tests and not disable them? Or just create a new issue for the unit test?

**Review by esigo — APPROVED** [review]
Thanks for the fix :)
changes look good just had one comment on the unit test

File: exporters/prometheus/test/exporter_utils_test.cc (hunk 1 above)
Comment (lalitb) [reply exporters/prometheus/test/exporter_utils_test.cc]: Thanks, have changed the unit tests now.

**Pull request merged**